File: Lista de Exercícios/Python/Configuracao.py
# Questão 14
class Configuracao:
    _instancia = None

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super().__new__(cls)
            cls._instancia.inicializar()
        return cls._instancia

    def inicializar(self):
        self.tema = "claro"
        self.idioma = "português"

    def set_tema(self, tema):
        self.tema = tema

    def set_idioma(self, idioma):
        self.idioma = idioma

    def get_configuracoes(self):
        return f"Tema: {self.tema}, Idioma: {self.idioma}"

# Exemplo de uso da classe Configuracao
config1 = Configuracao()
config1.set_tema("escuro")

config2 = Configuracao()
config2.set_idioma("inglês")

print(config1.get_configuracoes())
print(config2.get_configuracoes())
print(config1 is config2)

File: Lista de Exercícios/Golang/Configuracao.go
// Questão 14
package main

import (
	"fmt"
	"sync"
)

// Configuracao representa as configurações do sistema
type Configuracao struct {
	tema   string
	idioma string
}

var (
	instancia *Configuracao
	once      sync.Once
)

// GetInstancia retorna a única instância de Configuracao
func GetInstancia() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{
			tema:   "claro",
			idioma: "português",
		}
	})
	return instancia
}

// SetTema define o tema da configuração
func (c *Configuracao) SetTema(tema string) {
	c.tema = tema
}

// SetIdioma define o idioma da configuração
func (c *Configuracao) SetIdioma(idioma string) {
	c.idioma = idioma
}

// GetConfiguracoes retorna as configurações atuais
func (c *Configuracao) GetConfiguracoes() string {
	return fmt.Sprintf("Tema: %s, Idioma: %s", c.tema, c.idioma)
}

func main() {
	config1 := GetInstancia()
	config1.SetTema("escuro")

	config2 := GetInstancia()
	config2.SetIdioma("inglês")

	fmt.Println(config1.GetConfiguracoes())
	fmt.Println(config2.GetConfiguracoes())
	fmt.Printf("config1 e config2 são a mesma instância: %v\n", config1 == config2)
}
